app/grandchallenge/algorithms/templates/algorithms/job_form_update.html

```diff
@@ -3,7 +3,7 @@
 {% load url %}
 
 {% block title %}
-    Update Algorithm Result - {{ block.super }}
+    Update - {{ object.pk }} - {{ block.super }}
```

Comment:
Again, I think this doesn't follow the discussed pattern, right? "Update Job 1234 - Algorithm Foobar - GC" is what I would expect.